lldb/tools/lldb-dap/Handler/BreakpointLocationsHandler.cpp

```diff
@@ -71,22 +101,26 @@ BreakpointLocationsRequestHandler::Run(
       locations.emplace_back(line, column);
     }
   }
+}
 
-  // The line entries are sorted by addresses, but we must return the list
-  // ordered by line / column position.
-  std::sort(locations.begin(), locations.end());
-  locations.erase(llvm::unique(locations), locations.end());
+template <unsigned N>
+void BreakpointLocationsRequestHandler::AddAssemblyBreakpointLocations(
+    llvm::SmallVector<std::pair<uint32_t, uint32_t>, N> &locations,
```

Comment:
You can avoid the template by taking a reference to a `SmallVectorImpl<std::pair<uint32_t, uint32_t>>`, which `SmallVector` inherits from.

Reply:
Looking at how this is called, maybe you don't even need to take this by reference, and you can have this return a `llvm::SmallVector<std::pair<uint32_t, uint32_t>, 8>`. Honestly I don't even know if the `SmallVector` optimization is worth it here, personally I would've just returned a regular `std::vector`. If you change the return type, you can also change the name of the method from `Add` to `Get`. Now it looks like there's potentially multiple calls adding values to the location list.

Reply:
Agreem chganged to return `std::vector`